keyboards/skyloong/gk87/bl_pro/bl_pro.c

```diff
@@ -0,0 +1,192 @@
+//  Copyright 2023 NaturalZh (@NaturalZh)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include "quantum.h"
+#include "led_hc595.h"
+
+uint8_t IND = 0;  //buffer of LED Display
+
+int FN_ON = 0;
+bool WIN_LOCK = 0;
+bool DIS_BRETH = 0;
+bool SLEEP = 0;
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+       case MO(1):
+          if (record->event.pressed) {
+            FN_ON = 1;
+          } else {
+            FN_ON = 0;
+          }
+          return true;
+
+       case MO(2):
+           if (record->event.pressed) {
+             FN_ON = 1;
+           } else {
+             FN_ON = 0;
+           }
+          return true;
+
+       case MO(3):
+           if (record->event.pressed) {
+             FN_ON = 1;
+           } else {
+             FN_ON = 0;
+           }
+          return true;
+
+       case KC_LGUI:
+           if (FN_ON){
+              if ( record->event.pressed){
+                 WIN_LOCK = !WIN_LOCK ; //change win lock state
+                }
+              if (!WIN_LOCK) {
+                 IND = IND & (~WINLK_ON);   //Close win lock display
+                 s_serial_to_parallel(IND);
+                 gpio_write_pin_high(WIN_LOCK_PIN);
+                 return false; //windows key locked do nothing
+                }
+              s_serial_to_parallel(IND);
+            }
+           if (WIN_LOCK) {
+               IND = IND | WINLK_ON;   //Open win lock display
+               s_serial_to_parallel(IND);
+               gpio_write_pin_low(WIN_LOCK_PIN);
+               return false; //windows key locked do nothing
+            }
+           return true;  // continue all further processing of this key
+
+       case DF(0):
+           set_single_persistent_default_layer(0);
+           return true;
+
+       case DF(1):
+           set_single_persistent_default_layer(1);
+           return true;
+
```

Comment:
This still need to be removed.
```suggestion
```

Also setting this with the dip switch feature means you really don't need to set it as persistent, since that will always be called during init.  So this basically adds unnecessary writes to eeprom.